mitodl/mit-open
# 21: add keycloak authentication

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/mit-open/issues/21

#### What's this PR do?
Add the configuration properties required for facilitating authentication between MIT Open and Keycloak.

#### How should this be manually tested?

1. Using this PR's branch, visit: http://od.odl.local:8063/login/ol-oidc/
2. Add the following to your `.env` file
```
SOCIAL_AUTH_OL_OIDC_OIDC_ENDPOINT=https://sso-qa.odl.mit.edu/realms/olapps
OIDC_ENDPOINT=https://sso-qa.odl.mit.edu/realms/olapps
SOCIAL_AUTH_OL_OIDC_KEY=ol-open-local
SOCIAL_AUTH_OL_OIDC_SECRET=<REQUEST_FROM_PR_AUTHOR>
AUTHORIZATION_URL=https://sso-qa.odl.mit.edu/realms/olapps/protocol/openid-connect/auth
ACCESS_TOKEN_URL=https://sso-qa.odl.mit.edu/realms/olapps/protocol/openid-connect/token
USERINFO_URL=https://sso-qa.odl.mit.edu/realms/olapps/protocol/openid-connect/userinfo
```
This information is also retrievable from: https://sso-qa.odl.mit.edu/realms/olapps/.well-known/openid-configuration
3. You should be redirected to a login page.  Enter your Keycloak account credentials.
4. If your account credentials were correct, you should be redirected to the MIT Open index page.
5. If your MIT Open account has Django Admin access, you should be able to access the Django Admin interface at: http://od.odl.local:8063/admin/
6. If your MIT Open account has Django Admin access, you should be able to view the auth record for your user at: http://od.odl.local:8063/admin/social_django/usersocialauth/

#### Where should the reviewer start?

1. Request a Keycloak account from the PR author or DevOps member.
2. You have a corresponding MIT Open account.  Preferably, this account should have Django Admin access.

#### Any background context you want to provide?
The current Keycloak client that is configured for this PR should be updated to add more restrictive security around redirects and OpenID sources.
